### DISCUSSION TITLE
Want to get the old version of openfast

### DISCUSSION THREAD

**Discussion by playfulbluemoon** [OpenFAST/openfast#1546] [General]
Dear OpenFAST developer,

@jjonkman

Recently I was trying to run Dr.Lackner 's FAST-SC（It runs on the FAST V7 version.）, I found that I lacked some dat files, because the version of this FAST-SC is too old, so I can not find some of the necessary files on the Internet. I need the following documents, how should I find them, thank you very much.（Or where can I find a complete version of FAST V7）
NRELOffshrBsline5MW_Platform_ITIBarge4.dat
NRELOffshrBsline5MW_Tower_ITIBarge4.dat
NRELOffshrBsline5MW_Blade.dat
AeroDyn (v13.00.00a-bjj, 31-Mar-2010).

Best regards！

**Comment by jjonkman** [comment]
Dear @playfulbluemoon,

You can find a FAST v7.02 model of the NREL 5-MW baseline wind turbine atop the ITI Energy barge on my Google Drive: https://drive.google.com/file/d/1TaR2jmBnf3eah2bgrFvZhaJ02nJyzqjV/view?usp=share_link.

That said; is there are reason you need to run such an old version of the software?  The original structural control functionality added by Prof. Matt Lackner et al in FAST-SC has been migrated into OpenFAST (with improvements along the way) in what is now the StC submodel of ServoDyn; see the StC documentation here for more information: https://openfast.readthedocs.io/en/main/source/user/servodyn-stc/StC_index.html.  I suggest using StC within OpenFAST rather than using a now outdated version of the software.

Best regards,

**Comment by playfulbluemoon** [comment]
Dear @jjonkman，

Thank you for your patient answers and sharing, I am very grateful ! In fact, I am currently running Prof. Matt Lackner 's FAST-SC because it can directly input control force from simulink to TMD. I downloaded FAST-SC on the University of Massachusetts website and found that it could not be used directly.
![1682388913530](https://user-images.githubusercontent.com/126782805/234158544-1d5bba21-2353-472d-9075-4dfe3114c1c7.png)
This is my latest simulation results, it seems I lack a wind file.(I have added previously missing files，such as NRELOffshrBsline5MW_Platform_ITIBarge4.dat etc.）
Best regards,

**Reply by jjonkman** [reply]
Dear @playfulbluemoon,

The archive of the FAST v7.02 model of the NREL 5-MW baseline wind turbine atop the ITI Energy barge that I shared with you did not include a wind data file.  But you should able to generate your own wind data for use, e.g., using TurbSim.

Best regards,

**Reply by playfulbluemoon** [reply]
Dear @jjonkman，

Thank you very much for your advice, with your help I successfully run FAST-SC, but I still can not run simulink and FAST-SC co-simulation, I need to run make _ FAST _ SFunc.m.but it seems that I still lack some program files.
![048eabcb94857da4847a2d2b334575a](https://user-images.githubusercontent.com/126782805/234744577-cea8e3a4-e4c0-44a5-b6c8-b4a824f56b78.png)
![681e9954b5e3d68663f9af29f6b864b](https://user-images.githubusercontent.com/126782805/234744595-c215310e-3e75-4b5e-bb19-a137f34d6120.png)
Could you please tell me where i can find these files?
Best regards,

**Reply by jjonkman** [reply]
Dear @playfulbluemoon,

To compile FAST v7.02, you’ll need the source files from:
- FAST v7.02.00d-bjj: https://www.nrel.gov/wind/nwtc/assets/downloads/FAST/FASTv7.02/FAST_v7.02.00d-bjj.exe
- NWTC Library v1.07.00b-mlb: https://drive.google.com/file/d/11Evlt7vRaRQ_8u8QfRikkvnHj2aUOa2L/view?usp=sharing
- AeroDyn v13.00.02a-bjj: https://www.nrel.gov/wind/nwtc/assets/downloads/AeroDyn/AD_v13.00.02a-bjj.exe
- InflowWind v1.01.00b-bjj: https://drive.google.com/file/d/1W5okPCe3krvzg1ubgaX39PjaDanCsJqF/view?usp=sharing

Best regards,
